src/MarginTrading.SettingsService.AzureRepositories/Repositories/GenericAzureCrudRepository.cs

```diff
@@ -0,0 +1,89 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using AutoMapper;
+using AzureStorage;
+using AzureStorage.Tables;
+using Common.Log;
+using Lykke.SettingsReader;
+using MarginTrading.SettingsService.Core.Services;
+using MarginTrading.SettingsService.StorageInterfaces.Repositories;
+using Microsoft.WindowsAzure.Storage.Table;
+
+namespace MarginTrading.SettingsService.AzureRepositories.Repositories
+{
+    public class GenericAzureCrudRepository<TD, TE> : IGenericCrudRepository<TD>
+        where TD: class
+        where TE: SimpleAzureEntity, new()
+    {
+        protected readonly INoSQLTableStorage<TE> TableStorage;
+        protected readonly IConvertService ConvertService;
+        protected readonly ILog Log;
+
+        protected Action<IMappingOperationOptions<TD, TE>> DefaultAzureMappingOpts = opts => opts
+            .ConfigureMap(MemberList.Source).ForMember(e => e.ETag, e => e.UseValue("*"));
+        
+        protected readonly string PartitionKey;
+
+        protected GenericAzureCrudRepository(
+            ILog log,
+            IConvertService convertService,
+            IReloadingManager<string> connectionStringManager,
+            string tableName)
+        {
+            TableStorage = AzureTableStorage<TE>.Create(
+                connectionStringManager,
+                tableName,
+                Log,
+                new TimeSpan(1, 0, 0)
+            );
+
+            Log = log;
+            ConvertService = convertService;
+
+            PartitionKey = Activator.CreateInstance<TE>().PartitionKey;
+        }
+        
+        public virtual async Task<IReadOnlyList<TD>> GetAsync()
```

Comment:
If you use interfaces for entities, repo should return interfaces I think... or tell me the idea )